scripts/xtensa-build-zephyr.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2021 Intel Corporation. All rights reserved.
+
+# stop on most errors
+set -e
+
+SUPPORTED_PLATFORMS=(apl cnl icl tgl-h)
+BUILD_PATH=zephyr
+BUILD_ROM=no
+BUILD_JOBS=$(nproc --all)
+RIMAGE_KEY=modules/audio/sof/keys/otc_private_key.pem
+PLATFORMS=()
+
+PATH=$pwd/local/bin:$PATH
+
+pwd=$(pwd)
+
+die()
+{
+	>&2 printf '%s ERROR: ' "$0"
+	# We want die() to be usable exactly like printf
+	# shellcheck disable=SC2059
+	>&2 printf "$@"
+	exit 1
+}
+
+print_usage()
+{
+    cat <<EOF
+Re-configures and re-builds SOF with Zephyr using a pre-installed Zephyr toolchain and
+platform's _defconfig file.
+
+usage: $0 [options] platform(s)
+
+       -a Build all platforms
+       -j n Set number of make build jobs. Jobs=#cores by default.
+           Infinte when not specified.
+       -k Path a non-default rimage signing key.
+       -p Build path.
+
+Supported platforms ${SUPPORTED_PLATFORMS[*]}
+
+EOF
+}
+
+# parse the args
+while getopts "adj:k:p:ruv" OPTION; do
+        case "$OPTION" in
+		a) PLATFORMS=("${SUPPORTED_PLATFORMS[@]}") ;;
+		j) BUILD_JOBS=$OPTARG ;;
+		k) RIMAGE_KEY=$OPTARG ;;
+		p) BUILD_PATH=$OPTARG ;;
+		r) BUILD_ROM=yes ;;
+		*) print_usage; exit 1 ;;
+        esac
+done
+shift $((OPTIND-1))
```

Comment:
Please move most of/all the code to functions including a new, `main "$@"` function so:

- More variables can be `local` to reduce the risk of accidentally overwriting some user environment variables
- functions can be defined in any order, they don't need to be defined before being used
- moving code around functions does not change the indentation level which helps with git diff, rebase, cherry-pick, merge...
- it's easy to source the entire file without running it to test individual functions interactively